src/mame/drivers/1942.cpp

```diff
@@ -61,16 +61,20 @@ correctly.
 
 ***************************************************************************/
 
+namespace {
+
 /* 12mhz OSC */
-#define MASTER_CLOCK      (XTAL(12'000'000))
-#define MAIN_CPU_CLOCK      (MASTER_CLOCK/3)
-#define SOUND_CPU_CLOCK     (MASTER_CLOCK/4)
-#define AUDIO_CLOCK     (MASTER_CLOCK/8)
+constexpr auto MASTER_CLOCK	= 12_MHz_XTAL;
+constexpr auto MAIN_CPU_CLOCK	= MASTER_CLOCK/3;
+constexpr auto SOUND_CPU_CLOCK	= MASTER_CLOCK/4;
+constexpr auto AUDIO_CLOCK	= MASTER_CLOCK/8;
 /* 20mhz OSC - both Z80s are 4 MHz */
-#define MASTER_CLOCK_1942P     (XTAL(20'000'000))
-#define MAIN_CPU_CLOCK_1942P      (MASTER_CLOCK_1942P/5)
-#define SOUND_CPU_CLOCK_1942P     (MASTER_CLOCK_1942P/5)
-#define AUDIO_CLOCK_1942P     (MASTER_CLOCK_1942P/16)
+constexpr auto MASTER_CLOCK_1942P	= 20_MHz_XTAL;
+constexpr auto MAIN_CPU_CLOCK_1942P	= MASTER_CLOCK_1942P/5;
+constexpr auto SOUND_CPU_CLOCK_1942P	= MASTER_CLOCK_1942P/5;
+constexpr auto AUDIO_CLOCK_1942P	= MASTER_CLOCK_1942P/16;
+
+} // anonymous namespace
```

Comment:
Please only use `auto` type where one or more of the following applies:
* The type depends on template arguments
* The type is very deeply qualified and impractical to type out
* The type is a proxy/iterator to something obvious from context
* The type is a private inner type that you can't use directly by name

In this case, `auto` doesn't even save any characters vs `XTAL`.

Also, this stuff needs to be moved after the headers are included.  It shouldn't have been here in the first place as it messes with pre-compiled header behaviour.

Reply:
Fixed.